### PR TITLE
SQL-2455: Upgrade to commons-lang3 3.17.0

### PR DESCRIPTION
**Pull request by terakilobyte**
Sorry for the large diff...

### CODE REVIEW

**Review by nbagnard — CHANGES_REQUESTED** [review]
I don't think it's a blocking issue for users.
By default, Gradle will pull the highest version of a dependency if it is imported multiple times.
It also offers powerful [dependency constraint and conflict resolutions capabilities](https://docs.gradle.org/current/userguide/dependency_constraints_conflicts.html).

File: gradle.properties [hunk 1]
```diff
@@ -5,7 +5,7 @@ junitJupiterVersion = 5.5.2
 mockitoVersion = 3.0.0
 googleLintVersion = 1.7
 guavaVersion = 32.0.1-jre
-lang3Version = 3.0
+lang3Version=3.17.0
```
[the remark this answers is not in the document]
Comment (terakilobyte) [reply gradle.properties:8]: Works for me!

[the remark this answers is not in the document]
Comment (nbagnard) [reply gradle.properties]: Thank you :)
What about we do that for the other dependencies too.
This gives us free Java driver upgrade for example, isn't it?

[the remark this answers is not in the document]
Comment (terakilobyte) [reply gradle.properties]: It does, but I have another PR updating the driver that I plan to add this to (5.2.+)

**Review by EthanHardyMongo — APPROVED** [review]

**Pull request merged**